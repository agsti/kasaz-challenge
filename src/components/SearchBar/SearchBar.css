.search-container {
    display: flex;
    align-items: center;
    padding: 5px;
    background-color: #68388d;
}

.search-box {
    background-color: white;
    
    border-radius: 7px;
    border: 1px solid #aaa;

    height: 30px;
    width: 90%;

    display: flex;
    align-items: center;
}

.search-input {
    width: 100%;
    min-width: 0px;
    height: 30px;

    padding: 5px;
    border: none;
    background-color : transparent;
}

.search-icon {
    border-radius: 0px 7px 7px 0px;
    background-color: #41af68;
    margin-right: -0.5px;
    height: 100%;
}

.search-icon > img {
    width: 35px;
    height: 30px;
    padding: 6px;
}

.filter-btn {
    margin: 5px;
    width: 90px;
    text-align: center;
    color: white;
}

.filter-icon{
    width: 20px;
    height: 20px;
    margin-left: 5px;
    vertical-align: middle;
    transform: rotate(180deg);
}

.filter-icon.filter-menu-shown {
    transform: rotate(0deg);
}

