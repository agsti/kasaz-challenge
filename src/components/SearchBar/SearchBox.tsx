import React from "react";
import SearchIcon from "./icons/search-icon.svg";

type SearchInputProps = {
    onChange:  (arg0: React.ChangeEvent<HTMLInputElement>) => void,
    value: string,
    onSearch: (arg0: string) => void,
    showSearchButton: boolean
}

export default function SearchBox(props : SearchInputProps){
    const {
        onChange,
        value, 
        onSearch,
        showSearchButton
    } = props;

    return ( 
        <span className="search-box">
            <input 
            className="search-input"
            onChange = {onChange}
            value = {value}
            />
            {
            showSearchButton && <span
                className="search-icon"
            >
                <img src={SearchIcon} 
                    onClick={()=> onSearch(value)}
                />
            </span>
            }
        </span>
    )

}
