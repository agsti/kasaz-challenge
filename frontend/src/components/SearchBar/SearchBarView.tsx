import React, {useState} from "react";
import SearchBox from "./SearchBox";

import ArrowIcon from "./icons/arrow-icon.svg";

import "./SearchBar.css";

type SearchBarProps = {
    onSearch: (arg0: string) => void,
    searchValue: string,
    setSearchValue: (arg0: string) => void,
    searchHasChanged: boolean,
    onFilterButtonClicked: ()=>void,
    filterArrowLookingUp: boolean
}
export default function SearchBarView(props:SearchBarProps) {
    const {
        onSearch,
        setSearchValue, 
        searchValue,
        searchHasChanged,
        onFilterButtonClicked, 
        filterArrowLookingUp
    } = props

    return (
        <div className="search-container">
            <SearchBox
                onChange={(e) => {setSearchValue(e.target.value)}}
                value={searchValue}
                onSearch={onSearch}
                showSearchButton={searchHasChanged}
            />
            <div className="filter-btn" onClick={()=> onFilterButtonClicked()}>
                Filtros
                <img className={filterArrowLookingUp? "filter-icon filter-menu-shown" : "filter-icon"} src={ArrowIcon} />
            </div>
        </div>
    )
}
