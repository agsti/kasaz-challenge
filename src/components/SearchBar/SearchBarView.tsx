import React, {useState} from "react";
import SearchBox from "./SearchBox";

import ArrowIcon from "./icons/arrow-icon.svg";

import "./SearchBar.css";

type SearchBarProps = {
    onSearch: (arg0: string) => void,
    onFilterButtonClicked: ()=>void,
    filterMenushown: boolean
}
export default function SearchBarView(props:SearchBarProps) {
    const {onSearch, onFilterButtonClicked, filterMenushown} = props
    const [searchValue, setSearchValue] = useState("");
    return (
        <div className="search-container">
            <SearchBox
                onChange={(e) => {setSearchValue(e.target.value)}}
                value={searchValue}
                onSearch={onSearch}
            />
            <div className="filter-btn" onClick={()=> onFilterButtonClicked()}>
                Filtros
                <img className={filterMenushown? "filter-icon filter-menu-shown" : "filter-icon"} src={ArrowIcon} />
            </div>
        </div>
    )
}
