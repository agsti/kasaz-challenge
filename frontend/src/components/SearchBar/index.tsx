import React from "react";
import SearchBarView from "./SearchBarView";
import {useSelector, useDispatch} from "react-redux";

import {SelectMenuState} from "../../redux/Menus/selectors";
import {SetCity, SetFilterOutOfSync} from "../../redux/Filters/actions";
import {ToggleFilterMenu} from "../../redux/Menus/actions";
import { GetNewListings } from "../../redux/Listings/thunks";
import { SelectFilterState } from "../../redux/Filters/selectors";


export default function SearchBar() {
    const menuState = useSelector(SelectMenuState);
    const filterState = useSelector(SelectFilterState);
    const dispatch = useDispatch();

    return (
        <SearchBarView
            onSearch={search => {
                dispatch(GetNewListings())
                dispatch(SetFilterOutOfSync(false))
                dispatch(ToggleFilterMenu())
            }}
            onFilterButtonClicked={ () =>{
                dispatch(ToggleFilterMenu())
            }}
            filterArrowLookingUp={menuState.showFilterMenu}
            searchHasChanged={filterState.outOfSync}
            searchValue={filterState.city}
            setSearchValue={v => {
                dispatch(SetCity(v))
                !filterState.outOfSync && dispatch(SetFilterOutOfSync(true))
            }
            }
    />
    )
}
