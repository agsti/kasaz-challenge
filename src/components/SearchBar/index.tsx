import React from "react";
import SearchBarView from "./SearchBarView";
import {useSelector, useDispatch} from "react-redux";
import {SelectMenuState} from "../../redux/Menus/selectors";
import {SetCity} from "../../redux/Filters/actions";
import {ToggleFilterMenu} from "../../redux/Menus/actions";


export default function SearchBar() {
    const menuState = useSelector(SelectMenuState);
    const dispatch = useDispatch();

    return (
        <SearchBarView
            onSearch={search => {
                dispatch(SetCity(search))
            }}
            onFilterButtonClicked={ () =>{
                dispatch(ToggleFilterMenu())
            }}
            filterMenushown={menuState.showFilterMenu}
    />
    )
}
