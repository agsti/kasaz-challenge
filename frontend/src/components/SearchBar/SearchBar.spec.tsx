import React from 'react';
import { mount, shallow} from 'enzyme';
import {describe, expect, it} from '@jest/globals';

import SearchBar from './index';

describe('SearchBar', () => {
  it('should render correctly in "debug" mode', () => {
      const component = shallow(<SearchBar />);
  
    expect(component).toMatchSnapshot();
  });
});
