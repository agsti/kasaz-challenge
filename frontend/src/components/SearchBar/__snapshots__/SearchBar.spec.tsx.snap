// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBarView should render correctly with ArrowLooking Down 1`] = `
<div
  className="search-container"
>
  <SearchBox
    onChange={[Function]}
    onSearch={[MockFunction]}
    showSearchButton={false}
    value="search"
  />
  <div
    className="filter-btn"
    onClick={[Function]}
  >
    Filtros
    <img
      className="filter-icon"
      src=""
    />
  </div>
</div>
`;

exports[`SearchBarView should render correctly with ArrowLooking Up 1`] = `
<div
  className="search-container"
>
  <SearchBox
    onChange={[Function]}
    onSearch={[MockFunction]}
    showSearchButton={false}
    value="search"
  />
  <div
    className="filter-btn"
    onClick={[Function]}
  >
    Filtros
    <img
      className="filter-icon filter-menu-shown"
      src=""
    />
  </div>
</div>
`;
